### ISSUE TITLE
Training confusion

### ISSUE THREAD

**Issue by beyondbatman-master** [Microtreei/TSD-SR#7]
Have you encountered a situation where the output image shows periodic black spots and the training performance is poor? This issue occurred when switching the SD version from 2-1-base to 3-5. I would be very grateful if you could share some experience, 

![Image](https://github.com/user-attachments/assets/e4ebf2fa-77cb-4c51-8ea5-8cd18f8f2c2e)

**Comment by milely** [comment]
I encountered a similar problem. After switching from sd2.1 to sd3.5, the enlarged picture had obvious checkered block texture, and this was when I only used l2 loss and did not introduce vsd and tsm loss.

<img width="510" alt="Image" src="https://github.com/user-attachments/assets/d9a9b000-6271-4c54-a94b-c591520d543a" />

**Comment by milely** [comment]
The problem is not well solved, I am not sure what went wrong, there may be some training details that I did not notice, I may go back and recheck my training code.

**Comment by beyondbatman-master** [comment]
Oh oh, I'm also trying to solve this problem recently. Hope you can share some experiences if you have new findings, thanks.

**Comment by Microtreei** [comment]
To be honest, we had a similar situation when training our tsd-sr. We found that these spots may be due to the sd3 architecture. Similar problems exist when using sd3(28-step T2I default) to do 1～4 step inference. Our final solution is to adjust the learning rate(smaller) and balance the loss weight factors. It works, but it takes longer to train. By the way, we also conducted some experiments and found that using only l2 or lpips also takes a long time to train to remove these spots. This may be a problem that has not been solved well in training sd3 single-step models.

**Comment by milely** [comment]
Thank you for the author's response. Indeed, I have tried using the base model of SDXL and did not encounter the strange spots that appeared when using SD3. This issue is likely due to inherent problems with the SD3 model itself. I will attempt to implement the strategies you suggested to mitigate this problem.

**Comment by beyondbatman-master** [comment]
Thank you for author's reply. Maybe this is a problem with the DIT model structure, which is more sensitive to the weights of the losses. When I was experimenting with SD2, adjusting the weights of different losses didn't have much impact. However, in my recent experiments with the DIT model, if I adjust the loss weights, the final training curves show significant differences. I will try the strategies you suggested.

**Comment by ManuBN786** [comment]
Where did you guys find the training scripts ?

I can only see testing scripts

**Comment by ManuBN786** [comment]
Can you please provide the training scripts ?

**Comment by Microtreei** [comment]
We will provide the training code in the follow-up.

**Comment by zachary-zhexiao** [comment]
> We will provide the training code in the follow-up.

@Microtreei May I know when you can provide the training code? Thank you for your help!

**Comment by Microtreei** [comment]
We'll do it this month.

> > We will provide the training code in the follow-up.
> 
> [@Microtreei](https://github.com/Microtreei) May I know when you can provide the training code? Thank you for your help!

**Comment by EchoXu98** [comment]
> Have you encountered a situation where the output image shows periodic black spots and the training performance is poor? This issue occurred when switching the SD version from 2-1-base to 3-5. I would be very grateful if you could share some experience,
> 
> ![Image](https://github.com/user-attachments/assets/e4ebf2fa-77cb-4c51-8ea5-8cd18f8f2c2e)

There is something I don't understand about switching from sd2.1 to sd3.5. The author's model is based on the sd3 implementation. Do you mean to switch all the components about sd3 used in training to sd2? But that doesn't seem to be an easy thing to do.

**Comment by EchoXu98** [comment]
> I encountered a similar problem. After switching from sd2.1 to sd3.5, the enlarged picture had obvious checkered block texture, and this was when I only used l2 loss and did not introduce vsd and tsm loss.
> 
> <img alt="Image" width="510" src="https://private-user-images.githubusercontent.com/19663249/410283582-d9a9b000-6271-4c54-a94b-c591520d543a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIxNTA0MTcsIm5iZiI6MTc1MjE1MDExNywicGF0aCI6Ii8xOTY2MzI0OS80MTAyODM1ODItZDlhOWIwMDAtNjI3MS00YzU0LWE5NGItYzU5MTUyMGQ1NDNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzEwVDEyMjE1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxYmRjZWUxZjFjNDQ1MzdiODZjZTdiM2M2ZjY0YWQ5Y2M1YzRhMmZkN2EyNWU0OTNkZGQ1MGQwNWJmMWVmZmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.iFvpxbS8IAFuJrjvp_cA2xuf1gCplbKH3Orsdon5eRI">

There is something I don't understand about switching from sd2.1 to sd3.5. The author's model is based on the sd3 implementation. Do you mean to switch all the components about sd3 used in training to sd2? But that doesn't seem to be an easy thing to do.
